.github/actions/javascript/contributorChecklist/contributorChecklist.js

```diff
@@ -99,40 +100,39 @@ const completedContributorPlusChecklist = `- [x] I have verified the author chec
 const issue = github.context.payload.issue ? github.context.payload.issue.number : github.context.payload.pull_request.number;
 const combinedData = [];
 
-function printUncheckedItems(result) {
-    const checklist = result.split('\n');
+function getPullRequestBody() {
+    return GitHubUtils.octokit.pulls.get({
+        owner: GitHubUtils.GITHUB_OWNER,
+        repo: GitHubUtils.APP_REPO,
+        pull_number: issue,
+    }).then(({data: pullRequestComment}) => pullRequestComment.body);
+}
 
-    checklist.forEach((line) => {
-        // Provide a search string with the first 30 characters to figure out if the checkbox item is in the checklist
-        const lineSearchString = line.replace('- [ ] ', '').slice(0, 30);
-        if (line.includes('- [ ]') && (completedContributorChecklist.includes(lineSearchString) || completedContributorPlusChecklist.includes(lineSearchString))) {
-            console.log(`Unchecked checklist item: ${line}`);
-        }
-    });
+function getAllReviewComments() {
+    return GitHubUtils.paginate(GitHubUtils.octokit.pulls.listReviews, {
+        owner: GitHubUtils.GITHUB_OWNER,
+        repo: GitHubUtils.APP_REPO,
+        pull_number: issue,
+        per_page: 100,
+    }, response => _.map(response.data, review => review.body));
 }
 
-// Get all user text from the pull request, review comments, and pull request comments
-GitHubUtils.octokit.pulls.get({
-    owner: GitHubUtils.GITHUB_OWNER,
-    repo: GitHubUtils.APP_REPO,
-    pull_number: issue,
-}).then(({data: pullRequestComment}) => {
-    combinedData.push(pullRequestComment.body);
-}).then(() => GitHubUtils.octokit.pulls.listReviews({
-    owner: GitHubUtils.GITHUB_OWNER,
-    repo: GitHubUtils.APP_REPO,
-    pull_number: issue,
-})).then(({data: pullRequestReviewComments}) => {
-    pullRequestReviewComments.forEach(pullRequestReviewComment => combinedData.push(pullRequestReviewComment.body));
-})
-    .then(() => GitHubUtils.octokit.issues.listComments({
+function getAllComments() {
+    return GitHubUtils.paginate(GitHubUtils.octokit.issues.listComments, {
         owner: GitHubUtils.GITHUB_OWNER,
         repo: GitHubUtils.APP_REPO,
         issue_number: issue,
         per_page: 100,
-    }))
-    .then(({data: pullRequestComments}) => {
-        pullRequestComments.forEach(pullRequestComment => combinedData.push(pullRequestComment.body));
+    }, response => _.map(response.data, comment => comment.body));
+}
+
+getPullRequestBody()
```

Comment:
Should we use `Promise.all` to fire off `getPullRequestBody`, `getAllReviewComments`, and `getAllComments` at the same time?

Reply:
We could, but I don't see a huge benefit